cli/dts/lib.deno.ns.d.ts

```diff
@@ -184,7 +186,32 @@ declare namespace Deno {
    * });
    * ```
    * */
-  export function test(name: string, fn: () => void | Promise<void>): void;
+  export function test(name: string, fn: TestFunction): void;
+
+  export interface TestStep {
+    fn: TestFunction;
+    name: string;
+    ignore?: boolean;
+    /** Check that the number of async completed ops after the test step is the same
+     * as number of dispatched ops. Defaults to true.*/
+    sanitizeOps?: boolean;
+    /** Ensure the test sep does not "leak" resources - ie. the resource table
+     * after the test has exactly the same contents as before the test. Defaults
+     * to true. */
+    sanitizeResources?: boolean;
+
+    /** Ensure the test step does not prematurely cause the process to exit,
+     * for example via a call to `Deno.exit`. Defaults to true. */
+    sanitizeExit?: boolean;
+  }
+
+  export interface TestContext {
+    /** Execute a test step returning the error that occured if any.
```

Comment:
What is returned in case of no error? What if I `throw undefined`?

Reply:
> What is returned in case of no error? What if I `throw undefined`?

Was thinking we'll return the rejected value and type it as unknown.

Reply:
How do you differentiate between a passed and a failed test then?

Reply:
> How do you differentiate between a passed and a failed test then?

Can't if you reject with undefined but why would an assertion ever throw undefined? 🤔